### PR TITLE
Add option to disable cxx11 abi.

### PR DESCRIPTION
**Pull request by Shylock-Hg**

### CODE REVIEW

File: cmake/nebula/ABIConfig.cmake [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Is abi 11
```
Comment (laura-ding) [inline cmake/nebula/ABIConfig.cmake:1]: better move it to the client repo

File: cmake/nebula/InstallThirdParty.cmake [hunk 2]
```diff
@@ -9,13 +9,16 @@ if(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
 else()
     set(cxx_cmd ${CMAKE_CXX_COMPILER})
 endif()
+if(${DISABLE_CXX11_ABI})
```
Comment (laura-ding) [inline cmake/nebula/InstallThirdParty.cmake:12]: if used for `CMAKE_CXX_COMPILER_ID` is not Clang, move it to into else()

File: cmake/nebula/ABIConfig.cmake (hunk 1 above)
Comment (Shylock-Hg) [reply cmake/nebula/ABIConfig.cmake:1]: The common also must be compiled in abi98.

File: cmake/nebula/InstallThirdParty.cmake (hunk 2 above)
Comment (Shylock-Hg) [reply cmake/nebula/InstallThirdParty.cmake:12]: clang could disable cxx11 abi too.

**Review by dutor — APPROVED** [review]
LGTM.

**Review by laura-ding — APPROVED** [review]

File: cmake/nebula/ThirdPartyConfig.cmake [hunk 3]
```diff
@@ -5,16 +5,27 @@ message(">>>> Configuring third party for '${PROJECT_NAME}' <<<<")
...
-    elseif(NOT $ENV{NEBULA_THIRDPARTY_ROOT} STREQUAL "")
-        SET(NEBULA_THIRDPARTY_ROOT $ENV{NEBULA_THIRDPARTY_ROOT})
-    elseif(EXISTS /opt/vesoft/third-party)
-        SET(NEBULA_THIRDPARTY_ROOT "/opt/vesoft/third-party")
+
+if(${DISABLE_CXX11_ABI})
+    if(EXISTS ${CMAKE_BINARY_DIR}/third-party-98/install)
+        SET(NEBULA_THIRDPARTY_ROOT ${CMAKE_BINARY_DIR}/third-party-98/install)
     else()
+        SET(NEBULA_THIRDPARTY_ROOT ${CMAKE_BINARY_DIR}/third-party-98/install)
```
Comment (CPWstatic) [inline cmake/nebula/ThirdPartyConfig.cmake:13]:
```
if not exits ...
    include(InstallThirdParty)
endif()
set ...
```

Comment (Shylock-Hg) [reply cmake/nebula/ThirdPartyConfig.cmake]: Fixed.

**Review by laura-ding — APPROVED** [review]

**Review by dutor — APPROVED** [review]

**Pull request merged**